pynsot/commands/cmd_protocols.py

```diff
@@ -0,0 +1,464 @@
+        'running on.'
+    ),
+    required=True,
+)
+@click.option(
+    '-i',
+    '--id',
+    metavar='ID',
+    type=int,
+    help='Unique ID or natural key of the Protocol being updated.',
+)
+@click.option(
+    '-I',
+    '--interface',
+    metavar='INTERFACE',
+    type=types.NATURAL_KEY,
+    help=(
+        'The Interface this Protocol is running on. Either interface'
+        'or circuit must be populated.'
+    ),
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol is under.',
+    callback=callbacks.process_site_id,
+)
+@click.option(
+    '-t',
+    '--type',
+    metavar='TYPE',
+    type=types.NATURAL_KEY,
+    help='The Protocol type ID.',
+    required=True,
+)
+@click.option(
+    '--add-attributes',
+    'attr_action',
+    flag_value='add',
+    default=True,
+    help=(
+        'Causes attributes to be added. This is the default and providing it '
+        'will have no effect.'
+    )
+)
+@click.option(
+    '--delete-attributes',
+    'attr_action',
+    flag_value='delete',
+    help=(
+        'Causes attributes to be deleted instead of updated. If combined with'
+        'with --multi the attribute will be deleted if either no value is '
+        'provided, or if attribute no longer has an valid values.'
+    ),
+)
+@click.option(
+    '--replace-attributes',
+    'attr_action',
+    flag_value='replace',
+    help=(
+        'Causes attributes to be replaced instead of updated. If combined '
+        'with --multi, the entire list will be replaced.'
+    ),
+)
+@click.option(
+    '--multi',
+    is_flag=True,
+    help='Treat the specified attributes as a list type.',
+)
+@click.pass_context
+def update(ctx, attributes, auth_string, circuit, description, device, id, interface,
+           site_id, type, attr_action, multi):
+    """
+    Update a Protocol.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    You must provide the type of the protocol (e.g. OSPF, BGP, etc.)
+
+    When updating an Protocol you must provide the ID (-i/--id) and at least
+    one of the optional arguments.
+
+    The -a/--attributes option may be provided multiple times, once for each
+    key-value pair.
+
+    When modifying attributes you have three actions to choose from:
+
+    * Add (--add-attributes). This is the default behavior that will add
+    attributes if they don't exist, or update them if they do.
+
+    * Delete (--delete-attributes). This will cause attributes to be
+    deleted. If combined with --multi the attribute will be deleted if
+    either no value is provided, or if the attribute no longer contains a
+    valid value.
+
+    * Replace (--replace-attributes). This will cause attributes to
+    replaced. If combined with --multi and multiple attributes of the same
+    name are provided, only the last value provided will be used.
+    """
+    if not any([attributes, description, type]):
```

Comment:
You'll need to update this to include any field that can be updated (it's currently missing `auth_string`, `circuit`, `interface`).